### PR TITLE
Move CAClient interface defintion to the consumer KeyCertBundlerRotator.

### PR DESCRIPTION
**Pull request by incfly**
This makes adding methods to CAClient easier, no need to modify the fakeCAClient.

### CODE REVIEW

**Comment by incfly** [comment]
cc @wattli

File: security/pkg/caclient/client.go [hunk 1]
```diff
@@ -72,7 +67,7 @@ func NewCAClient(pltfmc platform.Client, ptclc grpc.CAGrpcClient, cAAddr string,
 
 // RetrieveNewKeyCert sends the CSR to Istio CA with automatic retries. When successful, it returns the generated key
 // and cert, otherwise, it returns error. This is a blocking function.
-func (c *cAClientImpl) RetrieveNewKeyCert() (newCert []byte, certChain []byte, privateKey []byte, err error) {
+func (c *CAClient) RetrieveNewKeyCert() (newCert []byte, certChain []byte, privateKey []byte, err error) {
```
Comment (myidpt) [inline security/pkg/caclient/client.go]: I think we need to modify the function signature to match Retrieve() (newCert, certChain, privateKey []byte, err error) to be able to use the interface KeyCertRetriever.

File: security/pkg/caclient/keycertbundlerotator_test.go [hunk 2]
```diff
@@ -20,27 +20,40 @@ import (
...
 )
 
+type fakeKeyCertRetriever struct {
+	NewCert    []byte
+	CertChain  []byte
+	PrivateKey []byte
+	Err        error
+}
+
+func (r *fakeKeyCertRetriever) Retrieve() (newCert, certChain, privateKey []byte, err error) {
```
Comment (myidpt) [inline security/pkg/caclient/keycertbundlerotator_test.go]: Is there any benefit of moving the mocks to here?

File: security/pkg/caclient/client.go (hunk 1 above)
Comment (incfly) [reply security/pkg/caclient/client.go]: Oh... Forgot this, fixed, thx!

File: security/pkg/caclient/keycertbundlerotator_test.go (hunk 2 above)
Comment (incfly) [reply security/pkg/caclient/keycertbundlerotator_test.go]: This test is the only one needs the fake, so I move it here & make it private. It reduces the public types we expose.

Comment (myidpt) [reply security/pkg/caclient/keycertbundlerotator_test.go]: OK. Thanks for clarification.

**Pull request merged**